apps/condo/domains/property/components/PropertyDocuments/PropertyDocuments.tsx

```diff
@@ -0,0 +1,234 @@
+import { OrganizationEmployeeRole, SortDocumentsBy } from '@app/condo/schema'
+import { Col, Row } from 'antd'
+import get from 'lodash/get'
+import omit from 'lodash/omit'
+import { useRouter } from 'next/router'
+import React, { useCallback, useMemo } from 'react'
+
+import { PlusCircle, Search } from '@open-condo/icons'
+import { useIntl } from '@open-condo/next/intl'
+import { ActionBar, Button, Select } from '@open-condo/ui'
+import { colors } from '@open-condo/ui/dist/colors'
+
+import Input from '@condo/domains/common/components/antd/Input'
+import { EmptyListContent } from '@condo/domains/common/components/EmptyListContent'
+import { DEFAULT_PAGE_SIZE, Table } from '@condo/domains/common/components/Table/Index'
+import { TableFiltersContainer } from '@condo/domains/common/components/TableFiltersContainer'
+import { useQueryMappers } from '@condo/domains/common/hooks/useQueryMappers'
+import { useSearch } from '@condo/domains/common/hooks/useSearch'
+import { getFiltersQueryData } from '@condo/domains/common/utils/filters.utils'
+import { getFiltersFromQuery, updateQuery } from '@condo/domains/common/utils/helpers'
+import { getPageIndexFromOffset, parseQuery } from '@condo/domains/common/utils/tables.utils'
+import { useUpdateDocumentModal } from '@condo/domains/document/hooks/useUpdateDocumentModal'
+import { useUploadDocumentsModal } from '@condo/domains/document/hooks/useUploadDocumentsModal'
+import { Document, DocumentCategory } from '@condo/domains/document/utils/clientSchema'
+import { usePropertyDocumentsTableColumns } from '@condo/domains/property/hooks/usePropertyDocumentsTableColumns'
+import { usePropertyDocumentsTableFilters } from '@condo/domains/property/hooks/usePropertyDocumentsTableFilters'
+
+
+const SORTABLE_PROPERTIES = ['name', 'category', 'createdAt']
+const DOCUMENTS_DEFAULT_SORT_BY = ['createdAt_DESC']
+
+const TableContent = ({ total, documentsLoading, propertyDocuments, openUploadModal, role, refetchDocuments }) => {
+    const intl = useIntl()
+    const AddDocumentMessage = intl.formatMessage({ id: 'documents.propertyDocuments.addDocument' })
+
+    const tableColumns = usePropertyDocumentsTableColumns()
+
+    const { UpdateDocumentModal, setSelectedDocument } = useUpdateDocumentModal()
+
+    const canManageDocuments = useMemo(() => get(role, 'canManageDocuments'), [role])
+
+    const handleRowAction = useCallback((document) => {
+        return {
+            onClick: async () => setSelectedDocument(document),
+        }
+    }, [setSelectedDocument])
+
+    return (
+        <>
+            <Row gutter={[0, 40]}>
```

Comment:
It's better to move lists outside in consts above, since it will be recreated (reallocated) for each rerender